runtime/src/bank.rs

```diff
@@ -2583,6 +2584,64 @@ impl Bank {
         debug!("Added native program {} under {:?}", name, program_id);
     }
 
+    /// Add a precompiled program
+    pub fn add_precompiled_program(&mut self, program_id: &Pubkey) {
```

Comment:
I think `add_native_program` isn't used for good reasons (maybe native program vs precompile program conceptual difference)...

how about adding a unit test for this new method (`add_precompiled_program`)?

Reply:
alternatively, I think enhancing add_native_program to be parameterized over precompile/native might be also desire to avoid code duplication...

Reply:
@ryoqun I think trying to combine those two functions would get too ugly and hard to read.  Instead I've pulled out some common code and cleaned them up.  Let me know what you think.  I also added more tests.

Reply:
got it